test/integration/discovery.go

```diff
@@ -251,6 +257,9 @@ var _ = Describe("Discover Cluster", func() {
 				if pod.GetPowerState() != proto.EntityDTO_POWERSTATE_UNKNOWN {
 					framework.Failf("All pods with NotReady Node provider should be in an unknown state")
 				}
+				if pod.GetPowerState() == proto.EntityDTO_POWERED_ON {
+					framework.Failf("All pods with NotReady Node provider should be in inactive state")
+				}
```

Comment:
I don't think this section is needed. We already fail the the test above if the power state is not `UNKNOWN`.